File: src/main/java/com/acevedosharp/entities/FamiliaDB.java
package com.acevedosharp.entities;

import javax.persistence.*;

@Entity
@Table(name = "familia", schema = "app")
public class FamiliaDB {
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private @Id @Column(name = "familia_id") Integer familiaId;
    private @Column(name = "nombre") String nombre;

    public FamiliaDB() {
    }

    public FamiliaDB(Integer familiaId, String nombre) {
        this.familiaId = familiaId;
        this.nombre = nombre;
    }

    public Integer getFamiliaId() {
        return familiaId;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }
}

File: src/main/kotlin/xyz/acevedosharp/persistence/entities/FamiliaDB.kt
package xyz.acevedosharp.persistence.entities

import xyz.acevedosharp.ui_models.Familia
import javax.persistence.*

@Entity
@Table(name = "familia", schema = "epic")
class FamiliaDB(

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "familia_id")
    var familiaId: Int? = null,

    @Column(name = "nombre")
    var nombre: String
) {
    fun toModel() = Familia(
        familiaId,
        nombre
    )

    override fun toString() = nombre
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as FamiliaDB

        if (familiaId != other.familiaId) return false
        if (nombre != other.nombre) return false

        return true
    }

    override fun hashCode(): Int {
        var result = familiaId ?: 0
        result = 31 * result + nombre.hashCode()
        return result
    }

}
